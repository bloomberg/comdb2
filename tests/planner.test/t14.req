explain query plan SELECT i.work_item_id FROM WorkItemAnalyst a join WorkItem i on i.work_item_id = a.work_item_id join Source s on s.source_id = i.source_id WHERE a.analyst_id = 14624 AND a.date_finished = -1 AND i.closed_on = -1 AND i.state_id = 2002 AND s.deleted_on = -1 and s.product_id in (18147,18148,18149) ORDER BY ( case i.work_item_type_id when 16166 then 0 when 16165 then 1000 else 9000 end + case i.work_item_sub_type_id when 20360 then 0 when 9 then 100 when 4 then 200 else 900 end + case when s.rank = 0 then 90 when s.rank >= 3 then s.rank+10 else s.rank end), i.created_on limit 1;
