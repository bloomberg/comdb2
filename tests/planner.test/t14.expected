(selectid=0, order=0, from=1, detail='SEARCH TABLE WorkItem AS i USING INDEX $FK_STATE_CDBB5048 (state_id=?) (7680 rows)')
(selectid=0, order=1, from=2, detail='SEARCH TABLE Source AS s USING COVERING INDEX $AK_SOURCE_ID_5BDAA304 (source_id=? AND deleted_on=?) (~1 row)')
(selectid=0, order=0, from=0, detail='EXECUTE LIST SUBQUERY 1')
(selectid=0, order=2, from=0, detail='SEARCH TABLE WorkItemAnalyst AS a USING COVERING INDEX $AK_WORKITEM_D4C11780 (work_item_id=? AND date_finished=? AND analyst_id=?) (~1 row)')
(selectid=0, order=0, from=0, detail='USE TEMP B-TREE FOR ORDER BY')
